erpnext/accounts/test/test_utils.py

```diff
@@ -73,6 +77,43 @@ def test_stock_voucher_sorting(self):
 		sorted_vouchers = sort_stock_vouchers_by_posting_date(list(reversed(vouchers)))
 		self.assertEqual(sorted_vouchers, vouchers)
 
+	def test_update_reference_in_payment_entry(self):
+		item = make_item().name
+
+		purchase_invoice = make_purchase_invoice(
+			**{"item": item, "supplier": "_Test Supplier USD", "currency": "USD", "conversion_rate": 82.32}
+		)
+		purchase_invoice.submit()
+
+		payment_entry = get_payment_entry(purchase_invoice.doctype, purchase_invoice.name)
+		payment_entry.target_exchange_rate = 62.9
+		payment_entry.paid_amount = 15725
+		payment_entry.deductions = []
+		payment_entry.insert()
+
+		self.assertEqual(payment_entry.difference_amount, -4855.00)
+		payment_entry.references = []
+
+		update_reference_in_payment_entry(
```

Comment:
@dj12djdjs In this test case you seem to be doing manually everything, an ideal test would be to make a standalone payment entry first and then reconcile using payment reconciliation and then check if the difference amount was updated or not